File: node/examples/xmlFormat.ts
/**
 * Content negotiation: fetch a law as XML instead of JSON.
 *
 * The typed resource methods always return JSON-parsed models. When your
 * app speaks XML, `requestRaw` fetches any endpoint in another wire
 * format and hands you the body untouched — the SDK ships no XML parser,
 * so parse `.text` with your own library.
 *
 * Usage:
 *
 *   LEGALIZE_API_KEY=leg_... npx tsx examples/xmlFormat.ts es BOE-A-1978-31229
 */

import { Legalize } from "@legalize-dev/sdk";

async function main(): Promise<number> {
  const country = process.argv[2] ?? "es";
  const lawId = process.argv[3] ?? "BOE-A-1978-31229";

  const client = new Legalize();
  try {
    // format defaults to "xml" → Accept: application/xml
    const res = await client.requestRaw("GET", `/api/v1/${country}/laws/${lawId}`);
    console.log(`status      ${res.statusCode}`);
    console.log(`contentType ${res.contentType}`);
    console.log("---");
    console.log(res.text.slice(0, 500)); // raw XML; parse with your own library

    // The same call with format: "json" gives you the JSON body to parse.
    const countries = await client.requestRaw("GET", "/api/v1/countries", { format: "json" });
    console.log("---");
    console.log(`countries payload is ${countries.contentType}`);
    console.log(countries.json());
  } finally {
    await client.close();
  }
  return 0;
}

main().then((code) => process.exit(code));

File: python/examples/xml_format.py
"""Fetch a law as XML via content negotiation.

The typed resource methods return JSON-parsed models; ``request_raw`` is
the escape hatch for callers whose pipeline speaks XML. It sets the
``Accept`` header for you and returns the body untouched.

Run:
    LEGALIZE_API_KEY=leg_... python examples/xml_format.py
"""

from __future__ import annotations

from legalize import Legalize


def main() -> None:
    with Legalize() as client:
        res = client.request_raw("GET", "/api/v1/es/laws/BOE-A-1978-31229")
        print("status:", res.status_code)
        print("content-type:", res.content_type)
        print(res.text[:400])

        # Parse it with the stdlib helper and pull a field out.
        root = res.xml()
        title = root.find("title")
        if title is not None:
            print("title:", title.text)


if __name__ == "__main__":
    main()
